project3/common/utils/utils.c

```diff
@@ -0,0 +1,66 @@
+#include "utils.h"
+
+long map_u(long x, long in_min, long in_max, long out_min, long out_max) {
```

Comment:
`_u` is probably not the best name for them. If the name is just `map` though there is a conflict because the `LiquidCrystal` library includes `Arduino.h` which defines `map` and `constrain` macros.